content/oci/oci.go

```diff
@@ -526,6 +514,72 @@ func (s *Store) GC(ctx context.Context) error {
 	return nil
 }
 
+// gcIndex reloads the index and updates metadata. Information of untagged blobs
+// are cleaned and only tagged blobs remain.
+func (s *Store) gcIndex(ctx context.Context) error {
+	// index tagged manifests
+	refMap := s.tagResolver.Map()
+	tagResolver := resolver.NewMemory()
+	graph := graph.NewMemory()
+	tagged := set.New[digest.Digest]()
+	for ref, desc := range refMap {
```

Comment:
nit: something like this might be more readable

```suggestion
	tagResolver := resolver.NewMemory()
	graph := graph.NewMemory()
	tagged := set.New[digest.Digest]()

	// index tagged manifests
	refMap := s.tagResolver.Map()
```